docs/developer/develop-minikube.md

```diff
@@ -0,0 +1,178 @@
+# Develop local with VSCode and Minikube
+
+## Introduction
+This document will describe how to setup a local development environment using VSCode as the IDE,  Minikube as the k8s cluster to run Turbinia and Skaffold to handle the development cycle. 
+
+This setup will provide run time debugging with breakpoints/watches in VSCode as well as hot-reloading of code changes to the live Turbinia deployment without having to rebuild the containers or restart the deployment.
+
+---
+NOTE: This setup has been tested and is in active use by the Turbinia developers in the following configurations.
+* MacAir M3 8GB with Docker Desktop
+* GCE (e2-standard-4) Debian 12/bookworm 16GB with Docker Engine (Running on GCP using the [VSCode Remote SSH extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh))
```

Comment:
How does this work with GCE then, is minikube still deployed locally on the GCE instance?  Are there any restrictions or special config to do things like attaching GCP disks?

Reply:
Yes, the GCE machine should be seen as a local machine in the cloud. The attaching of GCP disks is only possible in Turbinia when running under GKE due to the fact on how we connect disks to Nodes instead of Pods. The GKE development option will be added in a follow-up PR.

Reply:
I think attaching to a GCE instance will work, but not sure how running under minikube might change that.  I think we'd need to map in `/dev` if we're not already.   AFAIK we actually attach to the node and not the pod which is why we need the resource tracking code so we don't detach something already attached when a new worker/node is processing it.

https://github.com/google/turbinia/blob/1617dcc04c98e491f7c4f8ddeb260a3c1eb40559/turbinia/processors/google_cloud.py#L106
https://github.com/google/turbinia/blob/1617dcc04c98e491f7c4f8ddeb260a3c1eb40559/turbinia/evidence.py#L572

Reply:
Could be but I prefer a standard setup on all machines/environments for development and not customize it for different machines. If you want to develop by processing GCE disks the GKE development option is the way to go.